public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilters.test.tsx

```diff
@@ -66,10 +66,21 @@ describe('LabelFilters', () => {
 
   it('removes label', async () => {
     const { onChange } = setup({ labelsFilters: [{ label: 'foo', op: '=', value: 'bar' }] });
-    await userEvent.click(screen.getByLabelText(/remove/));
+    await userEvent.click(screen.getByLabelText(/remove-foo/));
     expect(onChange).toBeCalledWith([]);
   });
 
+  it('removes label but preserves a label with a value of empty string', async () => {
+    const { onChange } = setup({
+      labelsFilters: [
+        { label: 'foo', op: '=', value: 'bar' },
+        { label: 'le', op: '=', value: '' },
+      ],
+    });
+    await userEvent.click(screen.getByLabelText(/remove-foo/));
+    expect(onChange).toBeCalledWith([{ label: 'le', op: '=', value: '' }]);
+  });
+
```

Comment:
I think we need one more unit test that covers the case you described in the issue https://github.com/grafana/grafana/issues/78079

Reply:
There is just the one case, this test covers the case.

1. you have more than one label, here we have two labels, one has a vlaue of `""`, i.e. `{label_one="label_one_value", label_two=""}`
2. Using the query editor click x to delete `label_one`
3. The UI shows that the value from `label_one` has now been placed in `label_two`'s value like this: `{label_two="label_one_value"}`

You can see it in the video from the customer for the escalation
1. The customer has three labels, the last label has an empty string as a value
2. They delete the middle label
3. The value from the deleted label is now shown in the label with the `""` for a value. `""` has been replaced in the UI
https://drive.google.com/file/d/1rsi9Xb1Klra5f13AeOjXlnUdRXDuj3Gk/view

Reply:
@itsmylife It was just a UI bug where the LabelFilterItem needed to be re-rendered.